cmd/flags.go

```diff
@@ -83,8 +86,9 @@ func FlagGID(gid *int) {
 	flagSet.IntVar(gid, "gid", intEnv(EnvGID), "GID of user's group in the stack's build and run images")
 }
 
-func FlagGroupPath(path *string) {
-	flagSet.StringVar(path, "group", EnvOrDefault(EnvGroupPath, DefaultGroupPath), "path to group.toml")
+func FlagGroupPath(version, layersDir string, path *string) {
+	groupDir := getRelativePath(version, layersDir)
+	flagSet.StringVar(path, "group", EnvOrDefault(EnvGroupPath, filepath.Join(groupDir, DefaultGroupFileName)), "path to group.toml")
```

Comment:
Unfortunately this implementation doesn't work because at the the time `FlagGroupPath` is called, the the flags have not been parsed yet (see https://github.com/buildpacks/lifecycle/blob/main/cmd/command.go#L27-L38). There for `layerDir` string passed here and use to set the default is _always_ the default layers dir regardless of whether the user passes `-layers`. I think we need to either:
1. brute force this by not setting a default value for the flag and applying the computed default later if it is unset.
or
2. do something clever that I haven't through all the way with our command object that involve multiple rounds of parsing.